src/fiatconnect/ReviewScreen.tsx

```diff
@@ -117,6 +120,10 @@ export default function FiatConnectReviewScreen({ route, navigation }: Props) {
     )
   }
 
+  if (!showingExpiredQuoteDialog && normalizedQuote.getGuaranteedUntil() < new Date()) {
+    setShowingExpiredQuoteDialog(true)
```

Comment:
react question: even though this is gated by `showingExpiredQuoteDialog` being falsy, is it good practice to set state here? Wonder if we could just set initial state of `showingExpiredQuoteDialog` to `normalizedQuote.getGuaranteedUntil() < new Date()`? cc @kathaypacific

Reply:
yeah initial state sounds good. it is a little odd to have a possible state change on every render in the body of a component - state changes then cause another rerender and you could have a loop of these if not careful, so usually i'd expect state changes to be triggered inside callbacks or by other events. i think what is here is not dangerous, if it is just being used for the initial condition then putting it in where the `useState` is first declared is better, and then we'll only hit this computation once (on mount) :D